webapp/src/components/MarketplacePage/MarketplacePage.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import {
+  Menu,
+  Container,
+  Card,
+  Header,
+  Dropdown,
+  Pagination,
+  Loader,
+  Dimmer
+} from 'semantic-ui-react'
+import Publication from './Publication'
+import Navbar from 'components/Navbar'
+
+import { publicationType } from 'components/types'
+
+import {
+  SORT_TYPES,
+  getOptionsFromSortType,
+  getSortTypeFromOptions
+} from './utils'
+
+import './MarketplacePage.css'
+
+const sortOptions = Object.values(SORT_TYPES).map(type => ({
+  text: type,
+  value: type
+}))
+
+export default class MarketplacePage extends React.PureComponent {
+  static propTypes = {
+    publications: PropTypes.arrayOf(publicationType),
+    page: PropTypes.number.isRequired,
+    pages: PropTypes.number.isRequired,
+    sortBy: PropTypes.string.isRequired,
+    sortOrder: PropTypes.string.isRequired,
+    onNavigate: PropTypes.func.isRequired,
+    onConnect: PropTypes.func.isRequired,
+    onFetchPublications: PropTypes.func.isRequired
+  }
+
+  componentWillMount() {
+    const { onConnect, onFetchPublications } = this.props
+    onConnect()
+    onFetchPublications()
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { page, sortBy, sortOrder } = this.props
+    if (
+      page !== nextProps.page ||
+      sortBy !== nextProps.sortBy ||
+      sortOrder !== nextProps.sortOrder
+    ) {
+      this.shouldFetchPublications = true
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.shouldFetchPublications) {
+      this.props.onFetchPublications()
+      this.shouldFetchPublications = false
+    }
+  }
+
+  navigateTo(options) {
+    const navigationOptions = {
+      ...this.props,
+      ...options
+    }
+    const { page, sortBy, sortOrder, onNavigate } = navigationOptions
+    const url = `/marketplace?page=${page}&sort_by=${sortBy}&sort_order=${sortOrder}`
```

Comment:
Given that we have methods like `getPageFromRouter` that use `QueryString` on `./utils` can we do something like:

```javascript
import { buildFilterUrl } from './utils'

buildUrl(navigationOptions)

function buildUrl(navigationOptions) {
  return `${locations.marketplace}?${QueryString.stringify(navigationOptions)`
}
```

(not to be followed word-by-word, maybe unpacking is best to ensure certain params)

Reply:
I can't use `queryString.stringify` because the props and the query params use different casing: `sortBy` vs `sort_by`. But I can move that to utils and use `locations.marketplace` instead of `/marketplace`

Reply:
or i can use a `camelCase` to `snake_case` helper + `queryString.stringify`